include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp

```diff
@@ -111,10 +112,13 @@ class score_matrix_single_column
                 score_t const initial_value = score_t{})
     {
         this->number_of_columns = number_of_columns.get();
-        optimal_column.clear();
-        horizontal_column.clear();
```

Comment:
Mhmm, I really don't see how this code should be problematic. These are simply two std::vectors, nothing special.
Can you tell in which context this diagnostic error occurs, e.g. the test cases for which it does not compile?
Right now, I would keep the GCC workaround thing and maybe move it up here instead. This clearly indicates that something is weird with the compilers and that we don't know what the actual cause of this error is. Yet, a std::vector::clear should not be a problem at all.

Reply:
Moving the workaround to the resize also fixes both warnings

Reply:
I see. What I can tell from the compiler error is that the test is not built with SIMD instrutions enabled. Thus, there is this vector with aligned allocator defined as `aligned_allocator<__vector(1) int, 4>`. The last constant represents the byte alignment, which may cause some trouble here. Within the context of the SIMD alignment we would expect the type to have at least 16 byte alignment which would be the minimum for 128-bit SIMD operations. 
Maybe the whole thing can be fixed altogether if we choose for the alignment in the aligned_allocator definition something like: `std::max(alignof(value_t), alignof(std::max_align_t))`. 
At least for x86 architectures this would be at least 16 byte alignment.

Reply:
https://github.com/seqan/seqan3/blob/8c1a881e2fd7565f16dcd9efa7261711d7622836/include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp#L54

This would be changed to something like:

```cpp
static constexpr size_t align_of_score_v = std::min(alignof(score_t), alignof(std::max_align_t));
using physical_column_t = std::vector<score_t, aligned_allocator<score_t, align_of_score_v>>;

```

Reply:
I am curious whether this fixes it as well.

Reply:
Ohh, yes that makes sense. Explains the 1 VS 4 bytes. I'll try compiling with `-march=native` and no workaround to see whether it works. Then I'll try your typedef.